### PR TITLE
[Enhancement] support the CREATE EXTERNAL CATALOG xx（IF NOT EXISTS) syntax

### PR DESCRIPTION
**Pull request by zhaohehuhu**
## Why I'm doing:
Currently, Starrocks does't support the syntax, which would simplify operations for users by allowing them to create external catalogs more easily and robustly.

## What I'm doing:
make StarRocks support the CREATE EXTERNAL CATALOG xx IF NOT EXISTS syntax

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateCatalogStmt.java [hunk 1]
```diff
@@ -27,15 +27,20 @@ public class CreateCatalogStmt extends DdlStmt {
     private final Map<String, String> properties;
     private String catalogType;
 
-    public CreateCatalogStmt(String catalogName, String comment, Map<String, String> properties) {
-        this(catalogName, comment, properties, NodePosition.ZERO);
+    private final boolean ifNotExists;
+
+
+    public CreateCatalogStmt(String catalogName, String comment, Map<String, String> properties, boolean ifNotExists) {
+        this(catalogName, comment, properties, NodePosition.ZERO, ifNotExists);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateCatalogStmt.java:34]: better put the NodePosition.ZERO at the last argument

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 2]
```diff
@@ -207,6 +207,8 @@ public enum ErrorCode {
             "Unknown resource group '%s' "),
     ERR_BAD_CATALOG_AND_DB_ERROR(5080, new byte[] {'4', '2', '0', '0', '0'},
             "Unknown catalog.db '%s'"),
+    ERR_CATALOG_EXISTED_ERROR(5607, new byte[] {'4', '2', '0', '0', '0'},
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:210]: could you give me the error code refrence?

**Comment by imay** [comment]
@zhaohehuhu can you sign the CLA?

**Comment by zhaohehuhu** [comment]
> @zhaohehuhu can you sign the CLA?

Thanks. Done

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateCatalogStmt.java (hunk 1 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateCatalogStmt.java:34]: got it.

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 2 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:210]: Actually, it's no reference for this error code. I just created a new error code for it, which may be a bit unreasonable. Should I remove it and just throw an simple exception?

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 3]
```diff
@@ -207,6 +207,8 @@ public enum ErrorCode {
...
+            "catalog existed '%s'"),
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:211]: "Catalog '%s' already exists"

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 2 above)
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:210]: we just defined 5700-5799 as the external range. please use 5700 in the patch

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 3 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:211]: Got it. Thank.

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 2 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:210]: Got it. Thank.

Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java]: Done @stephen-shelby

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateCatalogStmt.java (hunk 1 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateCatalogStmt.java]: Done @Youngwb

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 3 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java]: catalog msg need to be consistent with db/table.

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 4]
```diff
@@ -221,6 +221,8 @@ public enum ErrorCode {
             "Export job [%s] not found when checking privilege"),
     ERROR_DYNAMIC_PARTITION_HISTORY_PARTITION_NUM_ZERO(5092, new byte[] {'4', '2', '0', '0', '0'},
             "Dynamic history partition num must greater than 0"),
+    ERR_CATALOG_EXISTED_ERROR(5700, new byte[] {'4', '2', '0', '0', '0'},
+            "catalog '%s' existed"),
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:225]:
```suggestion
            "Catalog '%s' already exists"),
```

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 3 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java]: OK. I missed it. Thank you!

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java (hunk 4 above)
Comment (zhaohehuhu) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java]: Changed. I didn't catch it earlier.

**Review by stephen-shelby — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
please add ut or sql test to cover your code.

**Comment by zhaohehuhu** [comment]
> please add ut or sql test to cover your code. Coverage requirement is 80%.

OK. I will add UTs later.

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by packy92** [comment]
please sign the commit. You could ref this [doc](https://dev.tobychung.com/signing-git-commits-with-ssh-keys)

**Pull request merged**